flare/kernels/map_3b_kernel.py

```diff
@@ -0,0 +1,654 @@
+        cut3b_mask (np.ndarray): nspec^2 long integer array
+        sig2: dummy
+        ls2: dummy
+        sig3 (np.ndarray): signal variances associates with three-body term
+        ls3 (np.ndarray): length scales associates with three-body term
+        cutoff_func (Callable): Cutoff function of the kernel.
+
+    Returns:
+        float:
+            Value of the 3-body force/energy kernel.
+    """
+
+    ej1 = etypes2[0]
+    ej2 = etypes2[1]
+    bc1 = spec_mask[c2]
+    bc2 = spec_mask[ej1]
+    bc3 = spec_mask[ej2]
+    ttype = triplet_mask[nspec * nspec * bc1 + nspec*bc2 + bc3]
+    ls = ls3[ttype]
+    sig = sig3[ttype]
+    r_cut = cutoff_3b
+
+    return three_body_mc_en_jit(env1.bond_array_3, env1.ctype,
+                                env1.etypes,
+                                env1.cross_bond_inds,
+                                env1.cross_bond_dists,
+                                env1.triplet_counts,
+                                c2, etypes2,
+                                r1, r2, r12,
+                                sig, ls, r_cut, cutoff_func) / 9.
+
+
+def three_body_mc_en_force(env1: AtomicEnvironment, r1, r2, r12, c2, etypes2,
+                           d1: int, hyps: 'ndarray', cutoffs: 'ndarray',
+                           cutoff_func: Callable = cf.quadratic_cutoff) \
+        -> float:
+    """3-body multi-element kernel between a force component and many local
+    energies on the grid.
+
+    Args:
+        env1 (AtomicEnvironment): First local environment.
+        rj1 (np.ndarray): matrix of the first edge length
+        rj2 (np.ndarray): matrix of the second edge length
+        rj12 (np.ndarray): matrix of the third edge length
+        c2 (int): Species of the central atom of the second local environment.
+        etypes2 (np.ndarray): Species of atoms in the second local
+            environment.
+        d1 (int): Force component of the first environment (1=x, 2=y, 3=z).
+        hyps (np.ndarray): Hyperparameters of the kernel function (sig1, ls1,
+            sig2, ls2).
+        cutoffs (np.ndarray): Two-element array containing the 2- and 3-body
+            cutoffs.
+        cutoff_func (Callable): Cutoff function.
+
+    Returns:
+        float:
+            Value of the 3-body force/energy kernel.
+    """
+    sig = hyps[0]
+    ls = hyps[1]
+    r_cut = cutoffs[1]
+
+    return three_body_mc_en_force_jit(env1.bond_array_3, env1.ctype,
+                                      env1.etypes,
+                                      env1.cross_bond_inds,
+                                      env1.cross_bond_dists,
+                                      env1.triplet_counts,
+                                      c2, etypes2,
+                                      r1, r2, r12,
+                                      d1, sig, ls, r_cut, cutoff_func) / 3
+
+def three_body_mc_en_force_sephyps(env1, r1, r2, r12, c2, etypes2,
```

Comment:
Phew... Multicomponent, energy/force, and separate hyps. Definitely a sign modularity is needed :)